psycop/common/model_training_v2/trainer/cross_validator.py

```diff
@@ -1,30 +1,105 @@
+import pandas as pd
+import polars as pl
+from sklearn.model_selection import StratifiedGroupKFold
+
+from psycop.common.model_training_v2.config.baseline_registry import BaselineRegistry
 from psycop.common.model_training_v2.loggers.base_logger import BaselineLogger
+from psycop.common.model_training_v2.trainer.base_dataloader import BaselineDataLoader
 from psycop.common.model_training_v2.trainer.base_trainer import (
     BaselineTrainer,
     TrainingResult,
 )
 from psycop.common.model_training_v2.trainer.preprocessing.pipeline import (
     PreprocessingPipeline,
 )
-from psycop.common.model_training_v2.trainer.preprocessing.polars_frame import (
-    PolarsFrame,
-)
-from psycop.common.model_training_v2.trainer.task.base_task import (
-    BaselineTask,
-)
+from psycop.common.model_training_v2.trainer.task.base_metric import BaseMetric
+from psycop.common.model_training_v2.trainer.task.base_task import BaselineTask
 
 
+@BaselineRegistry.trainers.register("crossval_trainer")
 class CrossValidatorTrainer(BaselineTrainer):
     def __init__(
         self,
-        data: PolarsFrame,
-        outcome_col_name: str,
+        training_data: BaselineDataLoader,
+        training_outcome_col_name: str,
         preprocessing_pipeline: PreprocessingPipeline,
         task: BaselineTask,
-        n_splits: int,
+        metric: BaseMetric,
         logger: BaselineLogger,
+        n_splits: int = 5,
+        group_col_name: str = "dw_ek_borger",
     ):
-        ...
+        self.training_data = training_data.load()
+        self.training_outcome_col_name = training_outcome_col_name
+        self.preprocessing_pipeline = preprocessing_pipeline
+        self.task = task
+        self.metric = metric
+        self.n_splits = n_splits
+        self.group_col_name = group_col_name
+        self.logger = logger
 
     def train(self) -> TrainingResult:
-        ...
+        training_data_preprocessed = self.preprocessing_pipeline.apply(
+            data=self.training_data,
+        )
+
+        X = training_data_preprocessed.drop(
+            self.training_outcome_col_name,
```

Comment:
This assumes that only one outcome column is in the dataset. We probably want to protect ourselves against that assumption. E.g. if we run this with my T2D dataset, 4 of the outcome columns will be in the training data.

That's a step for preprocessing, though, not this PR. I'll create an issue.